nginx.tmpl

```diff
@@ -19,7 +19,62 @@
 			server 127.0.0.1 down;
 		{{ end }}
 	{{ end }}
-	
+
+{{ end }}
+
+{{ define "location" }}
+location {{ .Path }} {
+	{{ if eq .Proto "uwsgi" }}
+		include uwsgi_params;
+		uwsgi_pass {{ trim .Proto }}://{{ trim .Upstream }};
+	{{ else if eq .Proto "fastcgi" }}
+		root   {{ trim .Vhostroot }};
+		include fastcgi.conf;
+		fastcgi_pass {{ trim .Upstream }};
+	{{ else }}
+		proxy_pass {{ trim .Proto }}://{{ trim .Upstream }};
```

Comment:
Maybe you can add a URL rewrite here as to remove the path used for routing when invoking the orginial host, which should solve the issues from @dotansimha and @evertramos:
 
`rewrite ^{{ .Path }}/(.*) /$1 break;`

Reply:
Maybe it would be nice to have this enabled/disabled by a variable, like  a boolean `VIRTUAL_PATH_REWRITE`.

Reply:
Note that this _can't_ be solved using `vhost.d/host.example.com_location` since that will be included in all locations, which doesn't seem to matter practically since that path will never be reached on `location /`, but it could matter for other options.